### PR TITLE
Cargo fmt and other misc updates

### PR DESCRIPTION
**Pull request by 0xBEEFCAF3**
Got a working example using Wasabi. In the process of getting everything set up, I updated some docs, updated some dependency versions, gitignore, and setup a template config file. Hope others will find this useful

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -207,49 +250,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
...
-                None => wallet_amount = bitcoin::Amount::from_str_in(arg.to_str().expect("wallet amount not UTF-8"), bitcoin::Denomination::Satoshi)?,
+                Some(channel_amount) => {
+                    scheduled_channels.push(ScheduledChannel::from_args(arg, channel_amount))
+                }
+                None => {
+                    wallet_amount = bitcoin::Amount::from_str_in(
+                        arg.to_str().expect("wallet amount not UTF-8"),
+                        bitcoin::Denomination::Satoshi,
+                    )?
+                }
```
Comment (Kixunil) [inline src/main.rs:300]: Missing commas are confusing.

File: src/main.rs [hunk 2]
```diff
@@ -207,49 +250,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
...
+            .test_connections(&mut handler.client)
+            .await;
+
+        println!(
+            "bitcoin:{}?amount={}&pj=https://example.com/pj",
+            address,
+            scheduled_payjoin
+                .total_amount()
+                .to_string_in(bitcoin::Denomination::Bitcoin)
+        );
```
Comment (Kixunil) [inline src/main.rs:322]: Amount string needs to go to a variable to not blow up the call. (Also ideally we would bump `bitcoin` version once it's realeased and use the newly added `display_in` method.)

File: conf_dir/conf.template [hunk 3]
```diff
@@ -0,0 +1,4 @@
+bind_port = 3000
+lnd_address = "tcp://localhost:10009"
```
Comment (Kixunil) [inline conf_dir/conf.template:2]: I will have to study what `tcp:` does, probably unsecure connection. Please use https instead or explain why you used `tcp`

**Review by Kixunil — CHANGES_REQUESTED** [review]
Thanks for the PR and reporting success with Wasabi! There are some good ideas. Unfortunately you made some mistakes and I hate `rustfmt` formatting. Please avoid reformatting the code, I will try to configure rustfmt to produce something nicer later. I pointed out things I dislike so you could try it too but I'd prefer to do it later, in a separate PR.

Also, very important, never mix formatting commits with anything else. It makes it painful to review and it is probably no-go in other projects as well.

File: Cargo.toml [hunk 4]
```diff
@@ -13,14 +13,14 @@ test_paths = []
...
-tonic_lnd = "0.3.0"
+tonic_lnd = "0.4.0"
 tokio = { version = "1.7.1", features = ["rt-multi-thread"] }
 rand = "0.8.4"
 base64 = "0.13.0"
 serde = "1.0.126"
 serde_json = "1.0.64"
 serde_derive = "1.0.126"
-ln-types = { version = "0.1.2", features = ["serde", "parse_arg"] }
+ln-types = { version = "0.1.3", features = ["serde", "parse_arg"] }
```
[the remark this answers is not in the document]
Comment (DanGould) [reply Cargo.toml:23]: Yes that has been my experience

File: .gitignore [hunk 5]
```diff
@@ -1 +1,11 @@
-/target
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
+
+# These are backup files generated by rustfmt
+**/*.rs.backup
```
[the remark this answers is not in the document]
Comment (DanGould) [reply .gitignore]: second, I'm not sure what this is even after trying to duplicate

File: README.md [hunk 6]
```diff
@@ -72,9 +72,14 @@ In other words, your grandmother will be able to somewhat privately open a bunch
 5. Copy BIP21 from command line output and paste it into one of the supported wallets
 6. Confirm the transaction and pray it works
 
+## Dev set up
+
+0. Copy `conf_dir/conf.template` to `conf_dir/conf` and replace values
```
[the remark this answers is not in the document]
Comment (DanGould) [reply README.md]: afaik there's no man page that shows that yet

File: src/main.rs [hunk 7]
```diff
@@ -16,7 +16,7 @@ extern crate configure_me;
 configure_me::include_config!();
 
 #[cfg(not(feature = "test_paths"))]
-const STATIC_DIR: &str = "/usr/share/loptos/static";
+const STATIC_DIR: &str = "static";
```
[the remark this answers is not in the document]
Comment (DanGould) [reply src/main.rs]: agree. compile with `cargo build --features=test_paths`. use the feature as intended

File: src/main.rs [hunk 8]
```diff
@@ -207,49 +250,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let (config, mut args) = Config::including_optional_config_files(std::iter::empty::<&str>()).unwrap_or_exit();
-
-    let client = tonic_lnd::connect(config.lnd_address, &config.lnd_cert_path, &config.lnd_macaroon_path)
+    // let (config, mut args) =
+    //     Config::including_optional_config_files(std::iter::empty::<&str>()).unwrap_or_exit();
+    let (config, mut args) =
+        Config::including_optional_config_files(&["conf_dir/conf"]).unwrap_or_exit();
```
[the remark this answers is not in the document]
Comment (DanGould) [reply src/main.rs]: seems like there's no man page which would help find this piece of docs

**Pull request closed without merging**